include/fiction/algorithms/physical_design/ortho_ordering_network.hpp

```diff
@@ -0,0 +1,633 @@
+                                {
+                                    paint_node_and_edges(ctn, current_node, ctn.color_south);
+                                }
+                            }
+                        }
+                    }
+                    // 3. PI connected to node connected to PI
+                    else if (auto pi_pi = ctn.color_ntk.get_pi_to_pi(); get_index(pi_pi, pi) % 2 == 0)
+                    {
+                        if (const auto fc = fanins(ctn.color_ntk, current_node);
+                            (ntk.is_inv(fc.fanin_nodes[0]) && !ntk.is_inv(fc.fanin_nodes[1])) ||
+                            (!ntk.is_inv(fc.fanin_nodes[0]) && ntk.is_inv(fc.fanin_nodes[1])))
+                        {
+                            paint_node_and_edges(ctn, current_node, ctn.color_south);
+                        }
+                        else
+                        {
+                            paint_node_and_edges(ctn, current_node, ctn.color_east);
+                        }
+                    }
+                });
+        });
+
+    return ctn;
+}
+
+template <typename Lyt, typename Ntk>
+class orthogonal_ordering_network_impl
+{
+  public:
+    orthogonal_ordering_network_impl(const Ntk& src, const orthogonal_physical_design_params& p,
+                                     orthogonal_physical_design_stats& st) :
+            ntk{input_ordering_view{mockturtle::fanout_view{
+                inverter_substitution(fanout_substitution<mockturtle::names_view<technology_network>>(src))}}},
+            ps{p},
+            pst{st}
+    {}
+
+    Lyt run()
+    {
+        // measure run time
+        mockturtle::stopwatch stop{pst.time_total};
+        // compute coloring
+        const auto ctn = conditional_coloring(ntk);
+
+        mockturtle::node_map<mockturtle::signal<Lyt>, decltype(ctn.color_ntk)> node2pos{ctn.color_ntk};
+
+        // instantiate the layout
+        Lyt layout{{1, 1, 1}, twoddwave_clocking<Lyt>(ps.number_of_clock_phases)};
+
+        // reserve PI nodes without positions
+        auto pi2node = reserve_input_nodes(layout, ctn.color_ntk);
+
+        // first x-pos to use for gates is 1 because PIs take up the 0th column
+        tile<Lyt> latest_pos{1, 0};
+        // ordering network tracker
+        tile<Lyt> latest_pos_inputs{0, 0};
+
+        // Reserve new columns for inverters in the ordering network
+        if (ctn.color_ntk.nc_inv_flag())
+        {
+            latest_pos.x = latest_pos.x + ctn.color_ntk.nc_inv_num() + 1;
+        }
+        // new column inverter tracker
+        std::uint64_t insert_position_inv{1};
+
+#if (PROGRESS_BARS)
+        // initialize a progress bar
+        mockturtle::progress_bar bar{static_cast<uint32_t>(ctn.color_ntk.size()), "[i] arranging layout: |{0}|"};
+#endif
+        // Find multi_output_nodes
+        std::vector<mockturtle::node<Ntk>> my_out_nodes;
+        std::vector<mockturtle::node<Ntk>> multi_out_nodes;
```

Comment:
Why are these two containers needed? What does `my_out_nodes`? The name doesn't give it away.